### ISSUE TITLE
building mesa via debos taking long time

### ISSUE THREAD

**Issue by openedev** [go-debos/debos#516]
I'm building the latest mesa as part of debos as panthor need mesa-24.0.0 on rk3588 which I don't find any package available.

```
 - action: run
    description: Build mesa
    chroot: true
    command: |
      set -e

      git clone https://gitlab.freedesktop.org/mesa/mesa.git /mesa
      cd /mesa && mkdir build debian && cd build

      # configure
      meson -Dvulkan-drivers= -Dgallium-drivers=panfrost,swrast -Dlibunwind=false -Dprefix=/mesa/debain -Dplatforms=x11,wayland

      # build
      ninja -v install -j 16
```

Is it proper way of doing the latest package to be part of debos? if so it is taking more than 3 hours to build but when I built in target it took only 5 to 10 minutes.

Any idea?

**Comment by kprasadvnsi** [comment]
assuming your host machine running the debos is x86_64. The qemu emulation for arm64 is very slow on x86 machine. I suggest you run this same debos recipe on your rk3588 device. it will be faster that way.

**Comment by sjoerdsimons** [comment]
Yup debos is indeed simply doing emulation in the chroot environment; So building in it is... going to be slow :) Not a debos issues as it's really not recommended to use debos as a build env (if you want to build mesa for aarch64 on x86, you'll be much better of cross-compiling)
